src/aarch64/ffi.c

```diff
@@ -740,6 +755,12 @@ ffi_call_int (ffi_cif *cif, void (*fn)(void), void *orig_rvalue,
    * points just after the alloca with the curretn stack bounds.
    */
   stack = __builtin_cheri_address_set(__builtin_cheri_stack_get(), (ptraddr_t)stack);
```

Comment:
This actually seems pretty dodgy. Yes, when we call the closure we want CSP's bounds to cover the entire stack, but we don't want to be using an unbounded capability in here, since we really should be fitting everything inside the alloca...